### PR TITLE
Adds system versioning to user, org, site, funding_space tables

### LINKED ISSUE #1039: Add create/update timestamps to app metadata objects

### PR DESCRIPTION
**Pull request by jlhogan**
this way, we have ability to restore data to old state (very good since process for fixing any of these DB entities is i manually modify em), and timestamps about data creation/update (very good for telling more stories about roll out and go live)

closes #1039

### CODE REVIEW

File: src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+import { addSystemVersioningColumnsToTableQuery } from './queries/addTemporalTableSupport';
+import { dropDefaultConstraint } from './utils/dropDefaultConstraint';
+
+export class AddTemporalTableSupportAppMetdata1610564061121
+  implements MigrationInterface {
+  name = 'AddTemporalTableSupportAppMetdata1610564061121';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
```
Comment (jlhogan) [inline src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts:10]: Leftover kruft was removed from models but no migration to clean up was created

File: src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      `ALTER TABLE "child" DROP COLUMN "specialEducationServicesType"`
+    );
+    await dropDefaultConstraint(
```
Comment (jlhogan) [inline src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts:13]: Align DB schema with models -- we added this default to make sure we appropriately back-filled the `incomeNotDisclosed` column in prod. In reality, we don't want a false default there; creating the record will null value is a valid (but unacceptable) action. User/UI is responsible for ensuring true/false value is appropriately provided

File: src/migration/queries/addTemporalTableSupport.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ export const addHistorySchemaQuery = () => `
 
 // from https://docs.microsoft.com/en-us/sql/relational-databases/tables/creating-a-system-versioned-temporal-table?view=sql-server-ver15#adding-versioning-to-non-temporal-tables
 export const addSystemVersioningColumnsToTableQuery = (tableName: string) => `
-	ALTER TABLE ${tableName} ADD
+	ALTER TABLE [${tableName}] ADD
```
Comment (jlhogan) [inline src/migration/queries/addTemporalTableSupport.ts:9]: needed for `user` table, b/c user is a protected word in sql server

Comment (JordanGuinn) [reply src/migration/queries/addTemporalTableSupport.ts:9]: _Of course it is_

File: src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+import { addSystemVersioningColumnsToTableQuery } from './queries/addTemporalTableSupport';
+import { dropDefaultConstraint } from './utils/dropDefaultConstraint';
+
+export class AddTemporalTableSupportAppMetdata1610564061121
```
Comment (JordanGuinn) [inline src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts:5]: nit - looks like there's an `a` missing in `AppMetdata`, both in the class name and the title

File: src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    await queryRunner.query(
+      `ALTER TABLE "child" DROP COLUMN "specialEducationServicesType"`
+    );
+    await dropDefaultConstraint(
+      queryRunner,
+      'income_determination',
+      'incomeNotDisclosed'
+    );
+
+    await queryRunner.query(addSystemVersioningColumnsToTableQuery('user'));
```
Comment (JordanGuinn) [inline src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts:19]: Is there no `down` counterpart for the `addSystemVersioningColumnsToTableQuery` function?

File: src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts (hunk 2 above)
Comment (JordanGuinn) [reply src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts:13]: +100000 for DB schema matching ORM relations 🔥

**Review by JordanGuinn — APPROVED** [review]

File: src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts (hunk 5 above)
Comment (jlhogan) [reply src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts:19]: not at present. I wrote this and don't see us ever actually wanting to undo it. But if we feel there's a legit use case for having down migrations that actually run (ok, i see after writing that there obviously is), we can make it a todo to add one and update these down migrations.

File: src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts (hunk 4 above)
Comment (jlhogan) [reply src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts:5]: yeahhhhh..... don't know that i care.....

File: src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts (hunk 2 above)
Comment (bamader) [reply src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts:13]: _THIS_ 😎

**Review by bamader — APPROVED** [review]

File: src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts (hunk 5 above)
Comment (JordanGuinn) [reply src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts:19]: We should absolutely add a TODO.  I def doubt we'll ever _want_ to undo it, more about having the option if we're ever forced to

Comment (jlhogan) [reply src/migration/1610564061121-AddTemporalTableSupportAppMetdata.ts:19]: done https://github.com/ctoec/data-collection/issues/1056

**Pull request merged**